File: yeji/DP/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        answer = []
        for i in range(numRows):
            answer.append([])
            for j in range(i+1):
                if j == 0 or j == i:
                    answer[i].append(1)
                else:
                    answer[i].append(answer[i-1][j-1] + answer[i-1][j])
        return answer

"""
종종 for의 범위 설정을 잘못하는거 같다..
그래서 한 5번 틀린듯ㅜ
"""

File: yeji/DP/pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function (numRows) {
  let answer = [];

  for (let i = 0; i < numRows; i++) {
    answer.push([]);

    for (let j = 0; j < i + 1; j++) {
      if (j === 0 || j === i) {
        answer[i].push(1);
      } else {
        answer[i].push(answer[i - 1][j - 1] + answer[i - 1][j]);
      }
    }
  }
  return answer;
};
